spec/beef/extensions/websocket_hooked_browser_spec.rb

```diff
@@ -0,0 +1,106 @@
+# encoding: UTF-8
+require 'rest-client'
+require 'core/main/network_stack/websocket/websocket'
+require 'websocket-client-simple'
+
+RSpec.describe 'BeEF Extension WebSockets' do
+
+  before(:all) do
+    @config = BeEF::Core::Configuration.instance
+    @cert_key = @config.get('beef.http.https.key')
+    @cert = @config.get('beef.http.https.cert')
+    @port = @config.get('beef.http.websocket.port')
+    @secure_port = @config.get('beef.http.websocket.secure_port')
+    @config.set('beef.http.websocket.secure', true)
+    @config.set('beef.http.websocket.enable', true)
+  end
+
+  it 'can hook a browser with websockets' do
+    # start beef server
+    @config = BeEF::Core::Configuration.instance
+    #set config parameters
+    @config.set('beef.credentials.user', "beef")
+    @config.set('beef.credentials.passwd', "beef")
+    @username = @config.get('beef.credentials.user')
+    @password = @config.get('beef.credentials.passwd')
+    #load extensions, best practice is to reload as previous tests can potentially cause issues.
+    BeEF::Extensions.load
+    sleep 2
+    if @config.get('beef.module').nil?
+      puts "loading modules"
+      BeEF::Modules.load
+      sleep 2
+    end
+    #generate token for the api to use
+    BeEF::Core::Crypto::api_token
+
+    # load up DB
+    # Connect to DB
+    ActiveRecord::Base.logger = nil
+    OTR::ActiveRecord.migrations_paths = [File.join('core', 'main', 'ar-migrations')]
+    OTR::ActiveRecord.configure_from_hash!(adapter:'sqlite3', database:'beef.db')
+    
+    # Migrate (if required)
+    context = ActiveRecord::Migration.new.migration_context
+    if context.needs_migration?
+      puts "migrating db"
+      ActiveRecord::Migrator.new(:up, context.migrations, context.schema_migration).migrate
+    end
+
+    #start the hook server instance, for it out to track the pids for graceful closure
+    http_hook_server = BeEF::Core::Server.instance
+    http_hook_server.prepare
+    @pids = fork do
+      BeEF::API::Registrar.instance.fire(BeEF::API::Server, 'pre_http_start', http_hook_server)
+    end
+    @pid = fork do
+      http_hook_server.start
+    end
+    # wait for server to start
+    sleep 1
```

Comment:
I think that this should all be in your before(:all) block, as generally that's where pre-test configuration should be run.

Reply:
Thats a good point, i think you're right and itll reduce the size of the test. I dont usually prefer to load it and instead use multiple expects to help confirm error location but this situation doesnt really rely care for stuff like that.